### PR TITLE
feat(demo): Footer standardised, use cases, mandatory and optional elements with knobs - FRONT-820

### PR DESCRIPTION
**Pull request by planctus**
# PR description

Please drop a few lines about the PR: what it does, how to test it, etc.

## QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [ ] I have put the vanilla component as `devDependencies`
- [ ] I have put the specs package as `devDependencies`
- [ ] I have added the components directly used in the twig file (with `include` or `embed`) as `dependencies`
- [ ] My component is listed in `@ecl-twig/ec-components`'s `dependencies`
- [ ] My variables naming follow the guidelines (snake case for twig)
- [ ] I have provided tests
- [ ] I have provided documentation (for the "notes" tab)
- [ ] If my local `yarn.lock` contains changes, I have committed it
- [ ] I have given my PR the proper label (`pr: review needed` to indicate that I'm done and now waiting for a review ,`pr: wip` to indicate that I'm actively working on it ...)

### CODE REVIEW

File: src/ec/packages/ec-component-footer-standardised/footer-standardised.story.js [hunk 1]
```diff
@@ -1,29 +1,265 @@
...
+      data[5].content_before,
+      optionalGroupId
+    );
+    data[5].list_class_name = text(
+      'sections[5].list_class_name',
+      data[5].list_class_name,
+      optionalGroupId
+    );
+    data[5].links.forEach((link, index) => {
+      console.log(link);
```
Comment (papegaill) [inline src/ec/packages/ec-component-footer-standardised/footer-standardised.story.js]: Should be removed ;)

File: src/ec/packages/ec-component-footer-standardised/footer-standardised.story.js [hunk 2]
```diff
@@ -1,29 +1,265 @@
...
   .addDecorator(withCode)
   .addDecorator(withNotes)
-  .add('default', () => footerStandardised(data), {
-    notes: { markdown: notes, json: data },
-  });
+  .addDecorator(withKnobs)
+  .add(
+    'default',
+    () => {
+      button('With our without class names', classBtnToggler, useCasesGroup);
```
Comment (papegaill) [inline src/ec/packages/ec-component-footer-standardised/footer-standardised.story.js]: With **or** without?

File: src/ec/packages/ec-component-footer-standardised/footer-standardised.story.js [hunk 3]
```diff
@@ -1,29 +1,265 @@
...
+      'sections[5].content_before',
+      data[5].content_before,
+      optionalGroupId
+    );
+    data[5].list_class_name = text(
+      'sections[5].list_class_name',
+      data[5].list_class_name,
+      optionalGroupId
+    );
+    data[5].links.forEach((link, index) => {
```
Comment (papegaill) [inline src/ec/packages/ec-component-footer-standardised/footer-standardised.story.js]: link declared, but never used on these foreach functions

File: src/ec/packages/ec-component-footer-standardised/footer-standardised.story.js (hunk 1 above)
Comment (planctus) [reply src/ec/packages/ec-component-footer-standardised/footer-standardised.story.js]: Mmhh..the question is: how does it come the linter did not protest? I noticed that in my environment the husky pre-commit was gone, but still we run the linter in the CI...

File: src/ec/packages/ec-component-footer-standardised/footer-standardised.story.js (hunk 2 above)
Comment (planctus) [reply src/ec/packages/ec-component-footer-standardised/footer-standardised.story.js]: :)

File: src/ec/packages/ec-component-footer-standardised/footer-standardised.story.js (hunk 3 above)
Comment (planctus) [reply src/ec/packages/ec-component-footer-standardised/footer-standardised.story.js]: Eh, but this is the parameter of a forEach:  function(currentValue, index, arr), if you need to access the index you also need the element, and that's what "link" is in this case. No issue from the linter, but i see we go even further :)

**Pull request closed without merging**